### PR TITLE
Improve styling of item-/option sets

### PR DESCRIPTION
**Pull request by alansemenov**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 13
- Added 9
           

Complexity increasing per file
==============================
- src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts  3
- src/main/resources/assets/admin/common/js/form/set/itemset/FormItemSetOccurrenceView.ts  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096&bid=20674427)

File: src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts [hunk 1]
```diff
@@ -19,23 +20,27 @@ export class FormOccurrenceDraggableLabel
...
-            nodes.push(document.createTextNode(' '));
-            let requiredMarker = new SpanEl('required');
-            nodes.push(requiredMarker.getHTMLElement());
-        }
         this.getEl().appendChildren(nodes);
     }
 
     setText(label: string) {
         this.title.nodeValue = label.trim();
+        this.toggleClass('custom-label', this.note !== label)
```
Comment (jsi) [inline src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts:43]: Codacy found an issue: [Missing semicolon](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/js/form/set/FormSetOccurrenceView.ts [hunk 2]
```diff
@@ -76,9 +76,9 @@ export class FormSetOccurrenceView
             return false;
         });
 
-        this.label = new FormOccurrenceDraggableLabel(this.getFormSet().getLabel(), this.getFormSet().getOccurrences());
+        this.label = new FormOccurrenceDraggableLabel(this.getFormSet().getLabel(), this.getFormSet().getOccurrences(), this.getFormSet().getLabel());
```
Comment (jsi) [inline src/main/resources/assets/admin/common/js/form/set/FormSetOccurrenceView.ts:79]: Codacy found an issue: [Exceeds maximum line length of 140](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 3]
```diff
@@ -1,207 +1,216 @@
...
-  > .occurrence-views-container > .form-item-set-occurrence-view {
-    padding-bottom: 20px;
-  }
-}
+    &.multiple-occurrence {
+      > .occurrence-views-container {
+        .clearfix();
 
-.form-item-set-view, .form-option-set-view {
+        > .form-item-set-occurrence-view, .form-option-set-occurrence-view {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:23]: Codacy found an issue: [Expected newline after "," (selector-list-comma-newline-after)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 4]
```diff
@@ -1,207 +1,216 @@
...
-  }
-}
+  .form-item-set-view, .form-option-set-view {
 
-.form-option-set-view .form-item-set-view {
-  > .occurrence-views-container > .form-item-set-occurrence-view {
-    padding-bottom: 20px;
-  }
-}
+    &.multiple-occurrence {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:19]: Codacy found an issue: [Unexpected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts [hunk 5]
```diff
@@ -19,23 +20,27 @@ export class FormOccurrenceDraggableLabel
...
+            const requiredMarker = new SpanEl('required');
+            nodes.push(requiredMarker.getHTMLElement());
+        }
+
         if (!!note) {
-            let noteEl = new Element(new NewElementBuilder().setTagName('sup').setGenerateId(true));
+            this.note = note;
+            const noteEl = new Element(new NewElementBuilder().setTagName('p').setGenerateId(true));
             noteEl.addClass('note');
+            noteEl.toggleClass('custom-label', this.note !== label)
```
Comment (jsi) [inline src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts:33]: Codacy found an issue: [Missing semicolon](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 6]
```diff
@@ -1,207 +1,216 @@
 @occurrence-label-margin-left: 10px;
+.form-view {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:2]: Codacy found an issue: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 7]
```diff
@@ -1,207 +1,216 @@
...
-      .form-option-set-occurrences-container {
-        padding-bottom: 10px;
-      }
+            &:after {
+              content: '*';
+              color: @input-red-text;
+            }
+          }
+
+          .drag-control {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:185]: Codacy found an issue: [Expected selector ".form-view .form-item-set-view .occurrence-views-container .form-item-set-occurrence-view .form-occurrence-draggable-label .drag-control" to come before selector ".form-view .form-item-set-view.multiple-occurrence > .occurrence-views-container > .form-item-set-occurrence-view > .form-occurrence-draggable-label > .drag-control" (no-descending-specificity)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 8]
```diff
@@ -1,207 +1,216 @@
...
+    & > .validation-viewer {
+      background-color: @admin-bg-light-gray;
+      padding: 0 10px 10px 10px;
+      margin-left: 0 !important;
     }
 
-  }
+    &.max-1-occurrence {
+      .occurrence-views-container {
+        > .form-option-set-occurrence-view {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:117]: Codacy found an issue: [Expected selector ".form-view .form-item-set-view.max-1-occurrence .occurrence-views-container > .form-option-set-occurrence-view" to come before selector ".form-view .form-item-set-view.multiple-occurrence > .occurrence-views-container .form-option-set-occurrence-view:not(:last-child)" (no-descending-specificity)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

File: src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less [hunk 9]
```diff
@@ -1,207 +1,216 @@
...
     }
 
-    .input-type-view,
-    .bottom-button-row,
-    .validation-viewer {
-      margin-left: @form-label-width;
-    }
+    .input-view {
+
+      &.invalid {
```
Comment (jsi) [inline src/main/resources/assets/admin/common/styles/api/form/set/form-set-view.less:52]: Codacy found an issue: [Unexpected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 16
           

Complexity increasing per file
==============================
- src/main/resources/assets/admin/common/js/form/FormOccurrenceDraggableLabel.ts  3
- src/main/resources/assets/admin/common/js/form/set/itemset/FormItemSetOccurrenceView.ts  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6371096&bid=20674427)

**Review by ashklianko — APPROVED** [review]

**Pull request merged**